code/modules/hydroponics/grown/misc.dm

```diff
@@ -1,19 +1,32 @@
-// Weeds
-/obj/item/seeds/weeds
-	name = "pack of weed seeds"
-	desc = "Yo mang, want some weeds?"
-	icon_state = "seed"
-	species = "weeds"
+// Starthistle
+/obj/item/seeds/starthistle
+	name = "pack of starthistle seeds"
+	desc = "A robust species of weed that often springs up in-between the cracks of spaceship parking lots"
+	icon_state = "seed-starthistle"
+	species = "starthistle"
 	plantname = "Starthistle"
-	lifespan = 100
+	lifespan = 70
 	endurance = 50 // damm pesky weeds
 	maturation = 5
 	production = 1
-	yield = -1
-	potency = -1
-	growthstages = 4
+	yield = 2
+	potency = 10
+	growthstages = 3
+	growing_icon = 'icons/obj/hydroponics/growing_flowers.dmi'
 	genes = list(/datum/plant_gene/trait/plant_type/weed_hardy)
+	mutatelist = list(/obj/item/seeds/harebell)
+
+/obj/item/seeds/starthistle/harvest(mob/user)
+	var/obj/machinery/hydroponics/parent = loc
+	var/seed_count = yield
+	if(prob(getYield() * 20))
+		seed_count++
+		var/output_loc = parent.Adjacent(user) ? user.loc : parent.loc
+		for(var/i in 1 to seed_count)
+			var/obj/item/seeds/starthistle/harvestseeds = src.Copy()
```

Comment:
I would. `src` is always implied, and therefore, not needed.

The only time you explicitly need to use `src` (aside form REALLY edge case BYOND bugs that most of us will never see) is if you have a proc that has arguments that are the same as that proc owner has vars:

```
/obj/widget/proc/copy_stuff(A, B, C)
	src.A = A
	src.B = B
	src.C = C
```
That's a case where it's ok to use `src`, but that's really not an argument for using `src` so much as it is avoiding arguments that are identical to vars on the proc owner.